string moveCommand
float64 goalX
float64 goalY
float64 speed
